bbq/vm/vm.go

```diff
@@ -1377,6 +1377,34 @@ func opDeref(vm *VM) {
 	vm.push(dereferenced)
 }
 
+func opStringTemplate(vm *VM, ins opcode.InstructionStringTemplate) {
+	expressions := vm.popN(int(ins.ExprSize))
+	values := vm.popN(int(ins.ValueSize))
+
+	// identical to interpreter
+	var builder strings.Builder
+	for i, str := range values {
+		s, ok := str.(*interpreter.StringValue)
+		if !ok {
+			panic(errors.NewUnreachableError())
+		}
+		builder.WriteString(s.Str)
+		if i < len(expressions) {
+			// switch on value instead of type
+			switch e := expressions[i].(type) {
+			case *interpreter.StringValue:
+				builder.WriteString(e.Str)
+			case interpreter.CharacterValue:
+				builder.WriteString(e.Str)
+			default:
+				builder.WriteString(e.String())
+			}
+		}
+	}
```

Comment:
Instead of copy-and-pasting this logic from `Interpreter.VisitStringTemplateExpression`, refactor the logic out of that method into a public function, and then call it from here.

That way the behaviour of the interpreter and VM stay identical when the feature is changed / it is less likely that the behaviour will become inconsistent.